model/map/src/main/java/org/keycloak/models/map/authorization/adapter/MapPolicyAdapter.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Override
+    public String getDescription() {
+        return entity.getDescription();
+    }
+
+    @Override
+    public void setDescription(String description) {
+        throwExceptionIfReadonly();
+        entity.setDescription(description);
+    }
+
+    @Override
+    public ResourceServer getResourceServer() {
+        return storeFactory.getResourceServerStore().findById(entity.getResourceServerId());
+    }
+
+    @Override
+    public Set<Policy> getAssociatedPolicies() {
+        String resourceServerId = entity.getResourceServerId();
+        return entity.getAssociatedPoliciesIds().stream()
+                .map(policyId -> storeFactory.getPolicyStore().findById(policyId.toString(), resourceServerId))
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public Set<Resource> getResources() {
+        String resourceServerId = entity.getResourceServerId();
+        return entity.getResourceIds().stream()
+                .map(resourceId -> storeFactory.getResourceStore().findById(resourceId.toString(), resourceServerId))
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public Set<Scope> getScopes() {
+        String resourceServerId = entity.getResourceServerId();
+        return entity.getScopeIds().stream()
+                .map(scopeId -> storeFactory.getScopeStore().findById(scopeId.toString(), resourceServerId))
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public String getOwner() {
+        return entity.getOwner();
+    }
+
+    @Override
+    public void setOwner(String owner) {
+        throwExceptionIfReadonly();
+        entity.setOwner(owner);
+    }
+
+    @Override
+    public void addScope(Scope scope) {
+        throwExceptionIfReadonly();
+        entity.addScope(UUID.fromString(scope.getId()));
+    }
+
+    @Override
+    public void removeScope(Scope scope) {
+        throwExceptionIfReadonly();
+        entity.removeScope(UUID.fromString(scope.getId()));
+    }
+
+    @Override
+    public void addAssociatedPolicy(Policy associatedPolicy) {
+        throwExceptionIfReadonly();
+        entity.addAssociatedPolicy(UUID.fromString(associatedPolicy.getId()));
+    }
+
+    @Override
+    public void removeAssociatedPolicy(Policy associatedPolicy) {
+        throwExceptionIfReadonly();
+        entity.removeAssociatedPolicy(UUID.fromString(associatedPolicy.getId()));
+    }
+
+    @Override
+    public void addResource(Resource resource) {
+        throwExceptionIfReadonly();
+        entity.addResource(UUID.fromString(resource.getId()));
+    }
+
+    @Override
+    public void removeResource(Resource resource) {
+        throwExceptionIfReadonly();
+        entity.removeResource(UUID.fromString(resource.getId()));
+    }
+
+    /**
+     * TODO: I am not sure what is the purpose of this method, seems it is not called at all, maybe we can remove?
+     */
+    @Override
+    public boolean isFetched(String association) {
```

Comment:
We can. Its purpose was related to caching but not associations are lazy loaded.